packages/insomnia/src/plugins/index.ts

```diff
@@ -311,6 +361,100 @@ export async function getTemplateTags(): Promise<TemplateTag[]> {
   return extensions;
 }
 
+export async function isPreBundlePluginTemplateTag(input: string) {
+  if (!input.includes('{%')) {
+    return false;
+  }
+  // const bunelPluginNames = getAppBundlePlugins().map(p => p.name);
+  // const activePlugins = await getActiveBundlePlugins();
+  const activeBundlePlugins = await getActiveBundlePlugins();
+  return activeBundlePlugins.some(plugin => {
+    const templateTags = plugin.module.templateTags || [];
+    const tagNames = templateTags.map(tt => tt.name);
+    return tagNames.some(tagName => input.match(new RegExp(`^{% *${tagName} *.+`)));
+  });
+}
+
+export async function getPluginRouterActions(): Promise<PluginAction[]> {
+  let extensions: PluginAction[] = [];
+
+  for (const plugin of await getActivePlugins()) {
+    const actions = plugin.module.pluginRouterActions || [];
+    extensions = [
+      ...extensions,
+      ...actions.map(p => ({
+        plugin,
+        ...p,
+      })),
+    ];
+  }
+
+  return extensions;
+}
+
```

Comment:
leave a comment over this clarifying it is only available to bundled plugins and is responsible for plugin code to execute a function in the ui component and ensure that once the response complete that the ui reloads?
```suggestion
// This is for insomnia code to reach out to bundled plugin functions
// It should only be available to bundled plugins
```